src/spi/graph/GraphClient.ts

```diff
@@ -63,28 +60,63 @@ export interface GraphClient {
      * @param {Q} variables the variables to be used
      * @param {any} options to be passed onto the underlying implementation
      * @returns {Promise<T>}
-     *
-     * DEPRECATED: use mutate() instead
+     * @deprecated use mutate() instead
      */
     executeMutation<T, Q>(mutation: string, variables?: Q, options?: any): Promise<any>;
 }
 
+/**
+ * Options to be passed into GraphClient.query.
+ */
 export interface QueryOptions<Q> {
+
+    /** Optional GraphQL query string */
     query?: string;
+    /**
+     * Optional absolute or relative path to the '.graphql' query file.
+     * Relative paths will get resolved against the path of the calling script.
+     */
     path?: string;
+    /**
+     * Optional name of the query operation in a '.graphql' file located in 'graphql/query'.
+     * The file system will be searched upwards for a 'graphql' folder; starting in
+     * the directory of the currently executing script.
+     */
     name?: string;
+    /** Optional directory containing GraphQL fragments; defaults to 'graphql/fragment' */
     fragmentDir?: string;
+    /** Optional variables passed to the GraphQL query */
     variables?: Q;
+    /** Optional options passed to the underlying implementation */
     options?: any;
 }
 
+/**
+ * Options to be passed to GraphClient.mutate.
+ */
 export interface MutationOptions<Q> {
+
+    /** Optional GraphQL mutation string */
     mutation?: string;
+    /**
+     * Optional absolute or relative path to the '.graphql' query file.
+     * Relative paths will get resolved against the path of the calling script.
+     */
     path?: string;
+    /**
+     * Optional name of the mutation operation in a '.graphql' file located in 'graphql/mutation'.
```

Comment:
One of `mutation` or `query` must be provided, correct?

Reply:
Add a interface-level comment to reflect that.